regression/goto-cc-cbmc-shared-options/object-bits/test.c

```diff
@@ -0,0 +1,30 @@
+#include <assert.h>
+#include <stdlib.h>
+
+extern __CPROVER_size_t __CPROVER_malloc_size;
+
+size_t
+find_first_set(const size_t max_malloc_size, const size_t bits_accumulator)
+{
+  // Implemented using recursion to avoid excessive unwinding or specifying an
+  // unwind limit.
```

Comment:
how does recursion stop excessive unwinding?

Reply:
I discovered this by experimentation. The same algorithm written as a loop spent a long time unwinding if no unwind limit was set. The recursive version only unwinds a sufficient number of times for the value of `max_malloc_size` which is actually specified.

I am not familiar with the unwinding code. However I would hazard a guess that loop unwinding happens early on, using a only a local analysis. Whereas recursion unwinding happens later, as part of symmex and the constant being passed to the function has been propagated in this case, leading to unwinding the exact number of times required.

Reply:
I haven't been able to reproduce the problem I saw with the iterative version and I never committed that version, so I can't get it back even with `git reflog`. Therefore I am going to assume there is no problem with unwinding loops / constant propagation. I shall delete this comment in order to avoid being misleading, but leave the recursive version in place because it doesn't seem worth rewriting this.